### PR TITLE
php81: replace local patch with an official commit

### PR DESCRIPTION
**Pull request by piotrkwiecinski**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/interpreters/php/8.1.nix [hunk 1]
```diff
@@ -21,7 +21,10 @@ let
...
       # Backport of PHP_LIBXML_IGNORE_DEPRECATIONS_START and PHP_LIBXML_IGNORE_DEPRECATIONS_END
       # Required for libxml2 2.13 compatibility patch.
-      ./php81-fix-libxml2-2.13-compatibility.patch
+      (fetchpatch {
+        url = "https://github.com/php/php-src/commit/e2d97314ab342d434e778cd00a2f34e4bdb07664.patch";
+        hash = "sha256-w0hyYUgbRGpvIBfWeDTSEUGpiJdyrtNjKy+Fn1vyAO0=";
+      })
       # Fix build with libxml2 2.13+. Has to be applied after libxml2 2.12 patch.
       (fetchpatch {
         url = "https://github.com/php/php-src/commit/9b4f6b09d58a4e54ee60443bf9a8b166852c03e0.patch";
```
Comment (jtojnar) [inline pkgs/development/interpreters/php/8.1.nix:30]: While at it, please replace this with https://github.com/php/php-src/commit/4fe821311cafb18ca8bdf20b9d796c48a13ba552 (the former is not on any branch)

Comment (piotrkwiecinski) [reply pkgs/development/interpreters/php/8.1.nix:30]: It is a backport to PHP-8.2 branch. See this PR https://github.com/php/php-src/pull/14816

Comment (jtojnar) [reply pkgs/development/interpreters/php/8.1.nix:30]: https://github.com/php/php-src/commit/9b4f6b09d58a4e54ee60443bf9a8b166852c03e0

> This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.

Comment (piotrkwiecinski) [reply pkgs/development/interpreters/php/8.1.nix:30]: Right..

Comment (piotrkwiecinski) [reply pkgs/development/interpreters/php/8.1.nix]: My apologies I missed it.

**Review by drupol — APPROVED** [review]

**Pull request merged**